File: Python/852. Peak Index in a Mountain Array.py
class Solution:
    def peakIndexInMountainArray(self, arr: List[int]) -> int:
#       return arr.index(max(arr))

        high, low = len(arr) - 1, 0
        while high > low:
            mid = (high + low) // 2
            if arr[mid] >arr[mid + 1]:
                high = mid
            else:
                low = mid + 1
        return low

File: C/852. Peak Index in a Mountain Array.c
int peakIndexInMountainArray(int* arr, int arrSize){
    int l = 0, r = arrSize, m;

    while (l <= r) {
        m = (l + r) / 2;

        if (*(arr + m) < *(arr + m + 1)) l = m + 1;
        else r = m - 1;
    }

    return l;
}
